### PR TITLE
initial solution

### PR DESCRIPTION
**Pull request by vageri**

### CODE REVIEW

File: src/test/java/com/epam/codingpractice/rangecompressor/RangeCompressorTest.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        input.add(new Range(6,10));
+
+        //WHEN
+        List<Range> compressedRanges = RangeCompressor.compressRanges(input);
+
+        //THEN
+        List<Range> expected = new LinkedList<>();
+        expected.add(new Range(1,5));
+        expected.add(new Range(6,10));
+        assertThat(compressedRanges,is(expected));
```
Comment (Zolikon) [inline src/test/java/com/epam/codingpractice/rangecompressor/RangeCompressorTest.java]: This is an incorrect test, does not match business requirements

File: src/test/java/com/epam/codingpractice/rangecompressor/RangeCompressorTest.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        List<Range> compressedRanges = RangeCompressor.compressRanges(input);
+
+        //THEN
+        List<Range> expected = new LinkedList<>();
+        expected.add(new Range(1,40));
+        assertThat(compressedRanges,is(expected));
+    }
+
+
+
```
Comment (Zolikon) [inline src/test/java/com/epam/codingpractice/rangecompressor/RangeCompressorTest.java]: I miss some test cases (which your solution does not handle at the moment)

File: src/test/java/com/epam/codingpractice/rangecompressor/RangeCompressorTest.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        input.add(new Range(4,15));
+        input.add(new Range(4,18));
+
+        //WHEN
+        List<Range> compressedRanges = RangeCompressor.compressRanges(input);
+
+        //THEN
+        List<Range> expected = new LinkedList<>();
+        expected.add(new Range(1,18));
+        assertThat(compressedRanges,is(expected));
```
Comment (Zolikon) [inline src/test/java/com/epam/codingpractice/rangecompressor/RangeCompressorTest.java]: This try of assertion gives unreadable message when it fails. Try it and try to solve it

File: src/test/java/com/epam/codingpractice/rangecompressor/RangeCompressorTest.java (hunk 1 above)
Comment (vageri) [reply src/test/java/com/epam/codingpractice/rangecompressor/RangeCompressorTest.java]: Oh, true. I'll fix this

**Pull request closed without merging**